File: frontend/src/client/models/BibleProgressStats.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ReadBookChapters } from './ReadBookChapters';

export type BibleProgressStats = {
    readTimes?: number;
    currentPercentage?: number;
    readBookChapters?: Array<ReadBookChapters>;
};

File: src/UseCases/Queries/GetUserBibleProgress/BibleProgressStats.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace UseCases.Queries.GetUserBibleProgress;

[DisplayName("BibleProgressStats")]
public record BibleProgressStats([Required] int ReadTimes, float CurrentPercentage, ICollection<ReadBookChapters> ReadBookChapters);

[DisplayName("ReadBookChapters")]
public record ReadBookChapters([Required] int BookId, [Required] ICollection<int> Chapters);